### ISSUE TITLE
Allow linking specific posts with =[url] or =[site] [id]

### ISSUE THREAD

**Issue by PatPeter** [AtoraSuunva/Booru-Discord#13]
I invited this bot to my Discord and it's working great, lots of entertainment value in #nsfw. Would it be possible to add functionality so that an ID or maybe even a URL can be passed in and BooruBot automatically expands it. As you know from coding it, Discord does not automatically preview Booru links. I don't know if just =[url] would work or if that would conflict with Discord.

If that won't work, maybe a way to specify the ID via =[site] [id], so =db 123456.

**Comment by AtoraSuunva** [comment]
I should be able to get both `=[url]` and `=[site] [id]` working, I'll add it to the list of stuff to do for the big rewrite i'm doing

**Comment by AtoraSuunva** [comment]
I'm planning to add `booru!link [link]` and have it be able to generate an embed from any link to a post

**Comment by PatPeter** [comment]
So the syntax would be this?

=db!link http://danbooru.donmai.us/post/show/123456

**Comment by AtoraSuunva** [comment]
The invoker in v3 is changing from `=` to `booru!` or `b!` (check #14 for details)

You'd link a post with either:

`booru!link db 252277`

or

`booru!link somebooru.com/link/to/post`

You can also be lazy and use the shorter aliases (`b!` for `booru!`, `ln` for `link`)

`b!ln boo.ru/link`

**Comment by AtoraSuunva** [comment]
Mini-update
`b!bd id:123456` works
http://i.imgur.com/gKPHlqJ.png

Still going to work on making `b!link http://danbooru.donmai.us/post/show/123456` grab the ID and post it however

**Comment by AtoraSuunva** [comment]
`b!link https://e621.net/post/show/1192588/ambiguous_gender-blue_eyes-blush-eeveelution-glace`
http://i.imgur.com/pxLttug.png
and
`b!link http://danbooru.donmai.us/posts/123456`
http://i.imgur.com/extRVKX.png

however `b!link http://danbooru.donmai.us/post/show/123456` doesn't work since booru shows the post view as `/posts/` and not `/post/show/`
Since `/post/show/` redirects to `/posts/` I'm gonna take the lazy route and not support `/post/show/` (also since it'd be a huge headache)

It was surprisingly simple actually (and horrifying)
http://i.imgur.com/nfhTjoX.png

tl;dr
use `b![site] id:[id]` if you have the site and id
use `b!link [url]` if you're too lazy to get the post id/site from the url yourself
Both create the same embeds
You can wrap the url in `<>` to disable the embed and it'll be fine

Commit: 2468b657a168ab624902dd4c000c8de314bb5dbe
